packages/venia-ui/lib/components/ScrollToTop/scrollToTop.js

```diff
@@ -0,0 +1,24 @@
+import React, { useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+
+// See: https://reacttraining.com/react-router/web/guides/scroll-restoration/scroll-to-top
+
+// Note: using { behavior: 'smooth' } in scrollTo() options often won't have an
+// effect because the re-render happens before smooth scrolling starts. Some
+// browsers try to do this internally but it's not universal.
+
+const ScrollToTop = () => {
```

Comment:
Now that we're migrated to function components this should be a peregrine hook, `useScrollTop`. It could even be `useScrollTopOnChange` and take a parameter like `pathname` which it uses in the effect. Then any component could configure when it should change such as `useScrollTopOnChange(user.isSignedIn)`, etc.

Could also have `useScrollTopOnMount` that scrolls only on mount of a component.

Reply:
I just saw Jimmy said basically the same thing a few days ago 😆